RHDEVDOCS-2330: Enhancements to starting pipelines via Topology (#42871)
openshift/openshift-docs: modules/op-starting-pipelines-from-topology-view.adoc

====
.Procedure

. Click *Topology* in the left navigation panel.
. Click the application to display *Pipeline Runs* in the side panel.
. In *Pipeline Runs*, click *Start Last Run* to start a new pipeline run with the same parameters and resources as the previous one. This option is disabled if a pipeline run has not been initiated. You can also start a pipeline run when you create it.
+
.Pipelines in Topology view
image::op_pipeline_topology.png[Pipelines in Topology view]
+
. In the *Topology* page, hover to the left of the application to see the status of its pipeline run. When the pipeline is added, the bottom left decorator indicates that there is an associated pipeline.
+
[NOTE]
====
The side panel of the application node in the *Topology* page displays a *Log Snippet* when a pipeline run fails on a specific task run. You can view the *Log Snippet* in the *Pipeline Runs* section, under the *Resources* tab. *Log Snippet* provides a general error message and a snippet of the log. A link to the *Logs* section provides quick access to the details about the failed run.
====

> . If a Pipeline run does not start automatically, the side panel displays a message that the pipeline cannot be automatically started. On the other hand, if a Pipeline is created but the user has not started the pipeline, its status is not started. When the user clicks the not started status icon, the start dialog box opens in the *Topology* view.

Reviewer: This is descriptive information, not a step a user can perform. Please remove the numbering. The step on line 24 also seems informational. 
Consider splitting everything from line 21 and down into a separate conceptual topic and present this information as a series of bullet points. What do you think?

Reviewer: Following the Mod Doc standard for content types is an important OKR. I think fixing this issue is important for us to comply with the Mod Docs standard.

Author: @rolfedh Can you please have a look and see if all looks correct?

// This module is included in the following assembly:
//
// *openshift_pipelines/working-with-pipelines-using-the-developer-perspective.adoc

:_content-type: PROCEDURE
[id="op-starting-pipelines_from_topology_view{context}"]
= Starting pipelines from Topology view

For pipelines created using the *From Git* option, you can use the *Topology* view to interact with pipelines after you start them:

[NOTE]
====
To see the pipelines created using the *Pipeline Builder* in the *Topology* view, customize the pipeline labels to link the pipeline with the application workload.
====
.Procedure

> To stary a pipeline in *Topology* view:

Reviewer 2: ```suggestion
To start a pipeline in *Topology* view:
```

> To start a pipeline in *Topology* view:

Reviewer: Avoid using introductory sentences in the procedures section.

> . On the left navigation panel, click *Topology*, and click on the application to see the pipeline runs listed in the side panel.

Reviewer: ```suggestion
. Click *Topology* in the left navigation panel.
. Click the application to see the pipeline runs listed in the side panel.
```

- One step per action. 
- Remove "on" following "click."

> . In the *Pipeline Runs* section, click *Start Last Run* to start a new pipeline run with the same parameters and resources as the previous one. This option is disabled if a pipeline run has not been initiated. You can also opt-in to starting the pipeline during the creation process itself.

Reviewer: ```suggestion
. In *Pipeline Runs*, click *Start Last Run* to start a new pipeline run with the same parameters and resources as the previous one. This option is disabled if a pipeline run has not been initiated. You can also start a pipeline run when you create it.
```

+

> .Pipelines in Topology view

Reviewer: Is this the beginning of a separate conceptual module?

Author: It's just a procedure which has been refactored inside a new one.

Author: @rolfedh This adds an image caption along with the figure number. Here is the preview: https://deploy-preview-42871--osdocs.netlify.app/openshift-enterprise/latest/cicd/pipelines/working-with-pipelines-using-the-developer-perspective#op-starting-pipelines_from_topology_viewusing-custom-pipeline-template-for-git-import

image::op_pipeline_topology.png[Pipelines in Topology view]
+
. In the *Topology* page, hover to the left of the application to see the status of the pipeline run for the application. While the pipeline is added, the bottom left decorator would indicate that there is an associated pipeline.
+
[NOTE]
====
The side panel of the application node in the *Topology* page displays a *Log Snippet* when a pipeline run fails on a specific task run. You can view the *Log Snippet* in the *Pipeline Runs* section, under the *Resources* tab. *Log Snippet* provides a general error message and a snippet of the log. A link to the *Logs* section provides quick access to the details about the failed run.
====

> . If a Pipeline cannot be automatically started, a message will be shown in the side panel stating that the pipeline cannot be automatically started. On the other hand, if a Pipeline is created but the user has not started the pipeline, it will have the status of not started. When the user clicks the not started status decorator, the start modal will open in the *Topology* view.

Reviewer: Hi Harsh. These are style recommendations from our various guides. But to save time, I'll summarize them rather than cite the sources:
- Avoid future tense ("will") in technical writing. Rather, use the present tense.
- "Modal" and " decorator" are jargon. Use the simpler "window" or "dialog box" instead.
- This isn't really a step. It seems like introductory information that you could break out into a series of optional if/then steps.
```suggestion
. If a Pipeline run does not start automatically, the side panel displays a message that the pipeline cannot be automatically started. On the other hand, if a Pipeline is created but the user has not started the pipeline, its status is not started. When the user clicks the not started status [icon?], the start [dialog box?] opens in the *Topology* view.
```suggestion
If a Pipeline cannot be automatically started, a message will be shown in the side panel stating that the pipeline cannot be automatically started. On the other hand, if a Pipeline is created but the user has not started the pipeline, it will have the status of not started. When the user clicks the not started status decorator, the start modal will open in the *Topology* view.
```

+
[NOTE]
====

> If the pipeline has no Build/Build Config, the Builds section should be hidden. If there is a Pipeline and Build Config, the Builds section would be displayed.

Reviewer: ```suggestion
If the pipeline has no Build or Build Config, the *Builds* section is not visible. If there is a Pipeline and Build Config, the Builds section is visible.
```
Avoid forward slashes. Replace with "and" or "or."
Avoid "would."

// This module is included in the following assembly:
//
// *openshift_pipelines/working-with-pipelines-using-the-developer-perspective.adoc

:_content-type: PROCEDURE
[id="op-starting-pipelines_from_topology_view{context}"]
= Starting pipelines from Topology view

For pipelines created using the *From Git* option, you can use the *Topology* view to interact with pipelines after you start them:

[NOTE]
====
To see the pipelines created using the *Pipeline Builder* in the *Topology* view, customize the pipeline labels to link the pipeline with the application workload.
====
.Procedure

. Click *Topology* in the left navigation panel.
. Click the application to display *Pipeline Runs* in the side panel.
. In *Pipeline Runs*, click *Start Last Run* to start a new pipeline run with the same parameters and resources as the previous one. This option is disabled if a pipeline run has not been initiated. You can also start a pipeline run when you create it.
+

> .Pipelines in Topology view

Reviewer: Is this really a procedure. It doesn't seem like one.

image::op_pipeline_topology.png[Pipelines in Topology view]
+
. In the *Topology* page, hover to the left of the application to see the status of its pipeline run. 

> . After a pipeline is added, a bottom left icon indicates that there is an associated pipeline.

Reviewer: This is not a user action. Should it be a step?

+
[NOTE]
====
The side panel of the application node in the *Topology* page displays a *Log Snippet* when a pipeline run fails on a specific task run. You can view the *Log Snippet* in the *Pipeline Runs* section, under the *Resources* tab. *Log Snippet* provides a general error message and a snippet of the log. A link to the *Logs* section provides quick access to the details about the failed run.
====

> . If a Pipeline run does not start automatically, the side panel displays a message that the pipeline cannot be automatically started. 

Reviewer: This is not a user action. Should it be a step?
